Character.cpp

```diff
@@ -0,0 +1,84 @@
+#include "Character.h"
+#include <iostream>
+#include <string>
+
+    //Constructor
+    Character::Character(const std::string characterName,const int characterHP, const int characterATK):name(characterName),HP(characterHP),ATK(characterATK){}
+
+    //Getters
+    std::string Character::getName() const
+    {
+        return this->name;
+    }
+
+
+    int Character::getHP() const
+    {
+        return this->HP;
+    }
+
+    int Character::getATK() const
+    {
+        return this->ATK;
+    }
+
+    //Convenience method for simple checking
+    bool Character::isDead() const {
+        return this->getHP() <= 0;
+    }
+
+    //Simple method for printing one's status
+    void Character::printStatus() const {
+        std::cout << this->getName()
+            + ": [HP: " << std::to_string(this->getHP())
+            + "] [DMG:" << std::to_string(this->getATK())
+            + "]" << std::endl;
+    }
+
+    //Instead of using an HP setter from outside of the class, we declare a 'sufferDamage' method
+    //The Character object's hp gets lowered based on the output 
+    void Character::sufferDamage(int atk) {
+        this->HP = (this->getHP() - atk);
+        if (this->HP < 0) { this->HP = 0; }
+    }
+
+    //Code piece responsible for handling a hit 
+    void Character::hit(Character* enemy) {
+        std::cout << this->getName() << " -> " << enemy->getName() << std::endl;
+        enemy->sufferDamage(this->getATK());
+    }
```

Comment:
Gondoljátok át, hogy szükség van-e a _hit_ ÉS a _sufferDamage_ függvényekre IS, vagy az egyiket esetleg el lehetne hagyni?